exercises/Stephen_Smith/ex02_oo_basics/include/Shapes.h

```diff
@@ -0,0 +1,218 @@
+#ifndef SHAPES_H_
+#define SHAPES_H_
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <math.h>  
+
+namespace{
+
+    const double PI=3.141592653589793238463;
+
+}
+
+
+class BaseShape{
+
+public:
+    
+    // Constructor
+    BaseShape();
+    virtual ~BaseShape() = 0 ;
```

Comment:
You are correct that we need to define this as virtual but it doesn't need to be abstract. We can define it as `virtual ~BaseShape() = default`